core/src/main/java/com/datastrato/gravitino/catalog/SchemaOperationDispatcher.java

```diff
@@ -159,47 +161,17 @@ public Schema createSchema(NameIdentifier ident, String comment, Map<String, Str
    */
   @Override
   public Schema loadSchema(NameIdentifier ident) throws NoSuchSchemaException {
-    NameIdentifier catalogIdentifier = getCatalogIdentifier(ident);
-    Schema schema =
-        doWithCatalog(
-            catalogIdentifier,
-            c -> c.doWithSchemaOps(s -> s.loadSchema(ident)),
-            NoSuchSchemaException.class);
+    EntityCombinedSchema schema =
+        TreeLockUtils.doWithTreeLock(ident, LockType.READ, () -> loadCombinedSchema(ident));
 
-    // If the Schema is maintained by the Gravitino's store, we don't have to load again.
-    boolean isManagedSchema = isManagedEntity(catalogIdentifier, Capability.Scope.SCHEMA);
-    if (isManagedSchema) {
-      return EntityCombinedSchema.of(schema)
-          .withHiddenPropertiesSet(
-              getHiddenPropertyNames(
-                  catalogIdentifier,
-                  HasPropertyMetadata::schemaPropertiesMetadata,
-                  schema.properties()));
+    if (schema == null || schema.imported()) {
+      return schema;
     }
 
-    StringIdentifier stringId = getStringIdFromProperties(schema.properties());
-    // Case 1: The schema is not created by Gravitino.
-    if (stringId == null) {
-      return EntityCombinedSchema.of(schema)
-          .withHiddenPropertiesSet(
-              getHiddenPropertyNames(
-                  catalogIdentifier,
-                  HasPropertyMetadata::schemaPropertiesMetadata,
-                  schema.properties()));
-    }
+    TreeLockUtils.doWithTreeLock(
+        NameIdentifier.of(ident.namespace().levels()), LockType.WRITE, () -> importSchema(ident));
 
-    SchemaEntity schemaEntity =
-        operateOnEntity(
-            ident,
-            identifier -> store.get(identifier, SCHEMA, SchemaEntity.class),
-            "GET",
-            stringId.id());
-    return EntityCombinedSchema.of(schema, schemaEntity)
-        .withHiddenPropertiesSet(
-            getHiddenPropertyNames(
-                catalogIdentifier,
-                HasPropertyMetadata::schemaPropertiesMetadata,
-                schema.properties()));
+    return schema;
```

Comment:
If schema is imported, we don't need to import it again, so we can return it directly.

Reply:
No, what I mean is that you will have a new entity combined schema in `importSchema()`, you should return the new one, not the old one. If you don't understand what I'm talking about, just ping me offline.

Reply:
Done.

Reply:
Has this comment been resolved? I have the same question, I think here should return `EntityCombinedSchema` after `import`.

Reply:
After digging into the code, the difference of `schema` before and after `import` is only about `AuditInfo`, and since the imported `entity` doesn't have the new `AuditInfo`, so `schema` is the same before and after `import`, no need to create a new one. It is also the same for table and topic. @mchades .

Reply:
Got it. If there is no change in the `auditInfo` then it is acceptable not to use imported `EntityCombinedSchema` here.